### DISCUSSION TITLE
[ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received an instance of Object

### DISCUSSION THREAD

**Discussion by educkf** [nuxt/nuxt#20420] [Questions] [answered]
Hello! 

I started facing an error suddenly on my Nuxt app, only when deploying to a Gcloud Function, there were no updated on the code base, it was kind of weird. On local development nothing wrong happens, it works just fine, 

My Nuxt version is `3.4.2` (but was happening also on `3.4.0`, I upgraded to see if it would solve)

The error is:

```
{
  "textPayload": "  err: TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received an instance of Object",
  "insertId": "644183b3000dd8f3640d5559",
  "resource": {
    "type": "cloud_function",
    "labels": {
      "project_id": "********",
      "region": "us-central1",
      "function_name": "server"
    }
  },
  "timestamp": "2023-04-20T18:25:55.907507Z",
  "labels": {
    "execution_id": "******3",
    "instance_id": "******"
  },
  "logName": "projects/*****/logs/cloudfunctions.googleapis.com%2Fcloud-functions",
  "trace": "projects/*****/traces/54c3b82432399866abd1ceea63ac7c5b",
  "receiveTimestamp": "2023-04-20T18:25:56.024201282Z"
}
```

The payload is a POST very similar to the $fetch example on documentation:

```
const data = await $fetch('/api/auth/login', {
    method: 'POST',
    body: JSON.stringify(body)
})
```

And the tracing points to the `readBody` composable, being used like this:

```
export default defineEventHandler(async (event) => {
    try {
        const body = await readBody(event)
        
        // (...)

    } catch (err) {
        console.log({ err })
        return AUTH_ERROR
    }
}
```

![image](https://user-images.githubusercontent.com/2439707/233456571-176d3c72-e69b-462f-b51d-91ee078f493c.png)

Any idea what it might be? I have tried already without `JSON.stringify()` on the body.

**Comment by educkf** [comment, marked as answer]
It was something with `$fetch`, changing to regular `fetch` solved the problem.

I have no idea why this happened, `$fetch` have already work before, and simplily stoped.

**Reply by vacijj** [reply]
I face the same problem with v3.4.2, but using `fetch` instead of `$fetch` did not help.

IMHO the issue should be reopened.

**Reply by vacijj** [reply]
We found the reason. See https://github.com/unjs/h3/pull/369

We fixed it with downgrading `h3` to `v1.6.2`. We added the following to the `package.json`:

```
{
  ...
  "dependencies": {
    ...
    "h3": "1.6.2"
  },
  "resolutions": {
    "h3": "1.6.2"
  }
}
```
